ddtrace/contrib/falcon/patch.py

```diff
@@ -24,8 +24,22 @@ def traced_init(wrapped, instance, args, kwargs):
     service = os.environ.get('DATADOG_SERVICE_NAME') or 'falcon'
     distributed_tracing = asbool(os.environ.get(
         'DATADOG_FALCON_DISTRIBUTED_TRACING')) or False
-
-    mw.insert(0, TraceMiddleware(tracer, service, distributed_tracing))
+    # It appears that the only way to install middleware is via the initializer
+    # so it's sufficient to check the middleware for our middleware to
+    # ensure idempotence.
+    # If the tracing middleware is already included then defer to that
+    if not _find_dd_middleware(mw):
```

Comment:
When writing this I considered the idea that if we find our middleware in the installed middlewares then we bump it to the front of the middlewares list.

However this may interfere with expected behaviour from the user when they are manually installing the middleware.

Reply:
Yeah, we should probably not do that. We should always try to inject the middleware into the front of the list when we can.

I think the long term direction of patching moves us away from middleware entirely anyways.